pgcli/main.py

```diff
@@ -203,11 +203,27 @@ def register_special_commands(self):
                                 'Execute commands from file.')
         self.pgspecial.register(self.write_to_file, '\\o', '\\o [filename]',
                                 'Send all query results to file.')
+        self.pgspecial.register(self.info_connection, '\\conninfo',
+                                '\\conninfo', 'Get connection details')
+
+    def info_connection(self, **_):
+        yield (None, None, None, 'You are connected to database "%s" as user '
+               '"%s" on host "%s" at port "%s".' % (self.pgexecute.dbname,
+                                                    self.pgexecute.user,
+                                                    self.pgexecute.host,
+                                                    self.pgexecute.port))
 
     def change_db(self, pattern, **_):
         if pattern:
-            db = pattern[1:-1] if pattern[0] == pattern[-1] == '"' else pattern
-            self.pgexecute.connect(database=db)
+            infos = re.findall(r'"[^"]*"|[^"\'\s]+', pattern)
+            infos.extend([None] * (4 - len(infos)))
+            map(lambda s: s.strip('"'), infos)
+            db, user, host, port = infos
+            try:
+                self.pgexecute.connect(database=db, user=user, host=host,
+                                       port=port)
+            except Exception as e:
+                print("Previous connection kept")
```

Comment:
You probably want to log this exception here also. And instead of `print`, better use `click.echo` (or `click.secho` if you want to style the error message).